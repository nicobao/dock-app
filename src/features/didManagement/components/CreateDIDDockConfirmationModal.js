import {Modal} from '../../../components/Modal';
import React from 'react';
import {translate} from '../../../locales';
import {LoadingButton, NBox, Typography} from '../../../design-system';
import {VStack} from 'native-base';
import {addTestId} from '../../../core/automation-utils';
import {AmountDetails} from '../../tokens/ConfirmTransactionModal';

export function CreateDIDDockConfirmationModal({
  didName,
  didType,
  onCreateDID,
  visible,
  onClose,
}) {
  const didTypeMap = {
    diddock: 'did:dock',
    didkey: 'did:key',
  };
  return (
    <Modal
      visible={visible}
      onClose={onClose}
      onBackButtonPress={onClose}
      modalSize={0.8}>
      <VStack p={7}>
        <Typography variant="h1" mb={4}>
          {translate('confirm_transaction.title')}
        </Typography>
        <NBox mt={5}>
          <Typography variant={'didModalInfoTitle'} mb={1}>
            {translate('didManagement.did_name')}
          </Typography>
          <Typography variant={'description'} mb={1}>
            {didName}
          </Typography>
        </NBox>

        <NBox mt={5}>
          <Typography variant={'didModalInfoTitle'} mb={1}>
            {translate('didManagement.did_type')}
          </Typography>
          <Typography variant={'description'} mb={1}>
            {didTypeMap[didType]}
          </Typography>
        </NBox>
        <NBox mt={5}>
          <Typography variant={'didModalInfoTitle'} mb={1}>
            {translate('confirm_transaction.fee')}
          </Typography>
          <AmountDetails amount={1} symbol={'DOCK'} />
        </NBox>
        <NBox mt={5}>
          <Typography variant={'didModalInfoTitle'} mb={1}>
            {translate('confirm_transaction.total')}
          </Typography>
          <AmountDetails amount={1} symbol={'DOCK'} />
        </NBox>
        <LoadingButton
          mt={12}
          full
          {...addTestId('CreateNewDIDScreenDIDCreate')}
          onPress={onCreateDID}>
          {translate('confirm_transaction.title')}
        </LoadingButton>
      </VStack>
    </Modal>
  );
}
