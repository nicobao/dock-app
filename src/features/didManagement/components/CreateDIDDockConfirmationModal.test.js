import React from 'react';
import {shallow} from 'enzyme';
import configureMockStore from 'redux-mock-store';
import {CreateDIDDockConfirmationModal} from './CreateDIDDockConfirmationModal';

const mockStore = configureMockStore();

describe('CreateDIDDockConfirmationModal', () => {
  it('should render correctly when visible', () => {
    const initialState = {
      // placeholder for redux store
    };

    const wrapper = shallow(
      <CreateDIDDockConfirmationModal
        visible={true}
        didName={'DIDName'}
        didType={'ed25519'}
        onCreateDID={jest.fn()}
        onClose={jest.fn()}
      />,
      {
        context: {store: mockStore(initialState)},
      },
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });
  it('should render correctly when not visible', () => {
    const initialState = {
      // placeholder for redux store
    };

    const wrapper = shallow(
      <CreateDIDDockConfirmationModal
        visible={false}
        didName={'DIDName'}
        didType={'ed25519'}
        onCreateDID={jest.fn()}
        onClose={jest.fn()}
      />,
      {
        context: {store: mockStore(initialState)},
      },
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
