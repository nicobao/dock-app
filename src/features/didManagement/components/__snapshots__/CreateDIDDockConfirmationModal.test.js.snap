// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateDIDDockConfirmationModal should render correctly when not visible 1`] = `
<ReactNativeModal
  animationIn="slideInUp"
  animationInTiming={300}
  animationOut="slideOutDown"
  animationOutTiming={300}
  avoidKeyboard={false}
  backdropColor="black"
  backdropOpacity={0.7}
  backdropTransitionInTiming={300}
  backdropTransitionOutTiming={300}
  coverScreen={true}
  customBackdrop={null}
  deviceHeight={null}
  deviceWidth={null}
  hasBackdrop={true}
  hideModalContentWhileAnimating={false}
  isVisible={false}
  onBackButtonPress={[MockFunction]}
  onBackdropPress={[MockFunction]}
  onModalHide={[Function]}
  onModalShow={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onSwipeComplete={[MockFunction]}
  panResponderThreshold={4}
  propagateSwipe={false}
  scrollHorizontal={false}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  style={
    Object {
      "justifyContent": "flex-end",
      "margin": 0,
    }
  }
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeDirection={
    Array [
      "down",
    ]
  }
  swipeThreshold={100}
  useNativeDriver={false}
>
  <View
    style={
      Object {
        "backgroundColor": "#18181B",
        "borderTopLeftRadius": 20,
        "borderTopRightRadius": 20,
        "height": 1067.2,
        "width": "100%",
      }
    }
  >
    <Memo(ForwardRef(Stack))
      alignItems="center"
      mt={6}
    >
      <Memo(ForwardRef(Box))
        backgroundColor="#3F3F46"
        borderRadius={6}
        height="4px"
        width="36px"
      />
    </Memo(ForwardRef(Stack))>
    <Memo(ForwardRef(VStack))
      p={7}
    >
      <Styled(Component)
        mb={4}
        variant="h1"
      >
        Confirm
      </Styled(Component)>
      <Memo(ForwardRef(Box))
        mt={5}
      >
        <Styled(Component)
          mb={1}
          variant="label"
        >
          DID Name
        </Styled(Component)>
        <Styled(Component)
          mb={1}
          variant="description"
        >
          DIDName
        </Styled(Component)>
      </Memo(ForwardRef(Box))>
      <Memo(ForwardRef(Box))
        mt={5}
      >
        <Styled(Component)
          mb={1}
          variant="label"
        >
          DID Type
        </Styled(Component)>
        <Styled(Component)
          mb={1}
          variant="description"
        />
      </Memo(ForwardRef(Box))>
      <Memo(ForwardRef(Box))
        mt={5}
      >
        <Styled(Component)
          mb={1}
          variant="label"
        >
          Transaction Fee
        </Styled(Component)>
        <Component
          amount={1}
          symbol="DOCK"
        />
      </Memo(ForwardRef(Box))>
      <Memo(ForwardRef(Box))
        mt={5}
      >
        <Styled(Component)
          mb={1}
          variant="label"
        >
          Total
        </Styled(Component)>
        <Component
          amount={1}
          symbol="DOCK"
        />
      </Memo(ForwardRef(Box))>
      <LoadingButton
        accessibilityLabel="CreateNewDIDScreenDIDCreate"
        full={true}
        mt={12}
        onPress={[MockFunction]}
        testID="CreateNewDIDScreenDIDCreate"
      >
        Confirm
      </LoadingButton>
    </Memo(ForwardRef(VStack))>
  </View>
</ReactNativeModal>
`;

exports[`CreateDIDDockConfirmationModal should render correctly when visible 1`] = `
<ReactNativeModal
  animationIn="slideInUp"
  animationInTiming={300}
  animationOut="slideOutDown"
  animationOutTiming={300}
  avoidKeyboard={false}
  backdropColor="black"
  backdropOpacity={0.7}
  backdropTransitionInTiming={300}
  backdropTransitionOutTiming={300}
  coverScreen={true}
  customBackdrop={null}
  deviceHeight={null}
  deviceWidth={null}
  hasBackdrop={true}
  hideModalContentWhileAnimating={false}
  isVisible={true}
  onBackButtonPress={[MockFunction]}
  onBackdropPress={[MockFunction]}
  onModalHide={[Function]}
  onModalShow={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onSwipeComplete={[MockFunction]}
  panResponderThreshold={4}
  propagateSwipe={false}
  scrollHorizontal={false}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  style={
    Object {
      "justifyContent": "flex-end",
      "margin": 0,
    }
  }
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeDirection={
    Array [
      "down",
    ]
  }
  swipeThreshold={100}
  useNativeDriver={false}
>
  <View
    style={
      Object {
        "backgroundColor": "#18181B",
        "borderTopLeftRadius": 20,
        "borderTopRightRadius": 20,
        "height": 1067.2,
        "width": "100%",
      }
    }
  >
    <Memo(ForwardRef(Stack))
      alignItems="center"
      mt={6}
    >
      <Memo(ForwardRef(Box))
        backgroundColor="#3F3F46"
        borderRadius={6}
        height="4px"
        width="36px"
      />
    </Memo(ForwardRef(Stack))>
    <Memo(ForwardRef(VStack))
      p={7}
    >
      <Styled(Component)
        mb={4}
        variant="h1"
      >
        Confirm
      </Styled(Component)>
      <Memo(ForwardRef(Box))
        mt={5}
      >
        <Styled(Component)
          mb={1}
          variant="label"
        >
          DID Name
        </Styled(Component)>
        <Styled(Component)
          mb={1}
          variant="description"
        >
          DIDName
        </Styled(Component)>
      </Memo(ForwardRef(Box))>
      <Memo(ForwardRef(Box))
        mt={5}
      >
        <Styled(Component)
          mb={1}
          variant="label"
        >
          DID Type
        </Styled(Component)>
        <Styled(Component)
          mb={1}
          variant="description"
        />
      </Memo(ForwardRef(Box))>
      <Memo(ForwardRef(Box))
        mt={5}
      >
        <Styled(Component)
          mb={1}
          variant="label"
        >
          Transaction Fee
        </Styled(Component)>
        <Component
          amount={1}
          symbol="DOCK"
        />
      </Memo(ForwardRef(Box))>
      <Memo(ForwardRef(Box))
        mt={5}
      >
        <Styled(Component)
          mb={1}
          variant="label"
        >
          Total
        </Styled(Component)>
        <Component
          amount={1}
          symbol="DOCK"
        />
      </Memo(ForwardRef(Box))>
      <LoadingButton
        accessibilityLabel="CreateNewDIDScreenDIDCreate"
        full={true}
        mt={12}
        onPress={[MockFunction]}
        testID="CreateNewDIDScreenDIDCreate"
      >
        Confirm
      </LoadingButton>
    </Memo(ForwardRef(VStack))>
  </View>
</ReactNativeModal>
`;
